develop/tutorials/articles/200-integrating-with-front-end-frameworks/01-integrating-with-react.markdown

```diff
@@ -0,0 +1,201 @@
+        	},
+        	"version": "1.0.0"
+        }    
+
+Next You can configure the portlet.
+
+## Configuring the Portlet [](id=configuring-the-portlet)
+ 
+Follow these steps to configure your portlet:
+
+1.  Create a Component class that implements the `Portlet.class` service:
+
+        @Component(
+        	immediate = true,
+        	property = {
+        		"com.liferay.portlet.display-category=category.sample",
+        		"com.liferay.portlet.instanceable=true",
+        		"javax.portlet.init-param.template-path=/",
+        		"javax.portlet.init-param.view-template=/view.jsp",
+        		"javax.portlet.name=" + MyNpmReactPortletKeys.MyNpmReact,
+        		"javax.portlet.resource-bundle=content.Language",
+        		"javax.portlet.security-role-ref=power-user,user"
+        	},
+        	service = Portlet.class
+        )
+        public class MyNpmReactPortlet extends MVCPortlet {
+          ...
+        }
+        
+2.  If your React project includes CSS styling as well, add the following 
+    additional property to specify the location of the main CSS file:
+    
+        "com.liferay.portlet.header-portlet-css=/css/main.css"
+
+    Note that this path is relative to the resources path. If using Sass, drop 
+    the `.scss` extension in this property and use `.css` instead. For example, 
+    if your main CSS file is located in 
+    `src/main/resources/META-INF/resources/css/app.scss`, then you would have 
+    the following configuration:
+    
+        "com.liferay.portlet.header-portlet-css=/css/app.css"
+
+3.  To improve code maintenance, 
+    [use the NPMResolver APIs to alias your module's package name](/develop/tutorials/-/knowledge_base/7-1/referencing-an-npm-modules-package). 
+    The example below exposes the module's name as `bootstrapRequire`:
+
+        @Override
+        public void doView(
+        		RenderRequest renderRequest, RenderResponse renderResponse)
+        	throws IOException, PortletException {
+
+        	JSPackage jsPackage = _npmResolver.getJSPackage();
+
+        	renderRequest.setAttribute(
+        		MyNpmReactWebKeys.BOOTSTRAP_REQUIRE,
+        		jsPackage.getResolvedId() + " as bootstrapRequire");
+
+        	super.doView(renderRequest, renderResponse);
+        }
+
+        @Reference
+        private NPMResolver _npmResolver;
+
+4.  Inside your `init.jsp`, add the following Java scriptlet to access the 
+    `bootstrapRequire` variable in your portlet's `view.jsp`:
+
+        <%
+        String bootstrapRequire = (String)renderRequest.getAttribute(
+          MyNpmReactWebKeys.BOOTSTRAP_REQUIRE
+        );
+        %>
+
+Next you can learn how to render your app's component and transpile your JS 
+files. 
+
+## Transpiling your JS and Rendering Your Component [](id=transpiling-your-js-and-rendering-your-component)
+
+Follow these steps to transpile your JS files and render your app component:
+
+1.  To use ES2015+ syntax in your portlet, you must transpile it for the 
```

Comment:
There's no need to change extensions. Just including the babel step in the package.json's build script is enough to transpile JS files.

Reply:
@izaera if the extension isn't needed, I was wondering why it's listed as `.es.js` here in the updated package.json file: https://github.com/liferay/liferay-blade-samples/blob/7.1/gradle/apps/npm/react-npm-portlet/package.json Should that be just `.js`, or is it transpiled to that format after babel runs?

Reply:
My guess is by no presets, you meant no liferay-npm-bundler presets. You still need the babel presets for babel I'm guessing.

Reply:
The extension is not relevant. I mean, you can use `.es.js`, `.js` or `.whatever.js`. The key thing is that the `.es` is not necessary. We can change the samples to make all the same if you want.

Reply:
The presets are always `devDependencies` because they are only used during the build, not runtime, and only by production code.

Reply:
You don't need any bundler preset at all (when using 2.x) because it comes built-in and we use one only smart preset for all configurations. 

Regarding babel, you only need babel presets in case you are doing something in your build that demands it. Otherwise, there's no need to configure any babel preset. I mean: we don't need babel for bundler related things because the bundler already calls babel internally when it needs it.

Reply:
Thanks @izaera !